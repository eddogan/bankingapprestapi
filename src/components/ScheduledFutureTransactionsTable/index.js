import React, { useState } from "react";
// Material-UI imports
import { makeStyles } from "@material-ui/core/styles";
import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  TableFooter,
  TablePagination
} from "@material-ui/core";
// Material-UI icon imports
import IconButton from "@material-ui/core/IconButton";
import FirstPageIcon from "@material-ui/icons/FirstPage";
import KeyboardArrowLeft from "@material-ui/icons/KeyboardArrowLeft";
import KeyboardArrowRight from "@material-ui/icons/KeyboardArrowRight";
import LastPageIcon from "@material-ui/icons/LastPage";
// Custom imports
import { formatStatusCode } from "../../helpers/FormatStatusCode";
import { formatCurrency } from "../../helpers/FormatCurrency";
import { formatDescription } from "../../helpers/FormatDescriptions";
import { formatDate } from "../../helpers/FormatDates";

function TablePaginationActions(props) {
  const classes = useStyles();
  const { count, page, rowsPerPage, onChangePage } = props;

  const handleFirstPageButtonClick = event => {
    onChangePage(event, 0);
  };

  const handleBackButtonClick = event => {
    onChangePage(event, page - 1);
  };

  const handleNextButtonClick = event => {
    onChangePage(event, page + 1);
  };

  const handleLastPageButtonClick = event => {
    onChangePage(event, Math.max(0, Math.ceil(count / rowsPerPage) - 1));
  };

  return (
    <div className={classes.root}>
      <IconButton
        onClick={handleFirstPageButtonClick}
        disabled={page === 0}
        aria-label="first page"
      >
        <FirstPageIcon />
      </IconButton>
      <IconButton
        onClick={handleBackButtonClick}
        disabled={page === 0}
        aria-label="previous page"
      >
        <KeyboardArrowLeft />
      </IconButton>
      <IconButton
        onClick={handleNextButtonClick}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="next page"
      >
        <KeyboardArrowRight />
      </IconButton>
      <IconButton
        onClick={handleLastPageButtonClick}
        disabled={page >= Math.ceil(count / rowsPerPage) - 1}
        aria-label="last page"
      >
        <LastPageIcon />
      </IconButton>
    </div>
  );
}

export default function ScheduledFutureTransactionsTable(props) {
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);

  const handleChangePage = newPage => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = event => {
    setRowsPerPage(parseInt(event.target.value, 10));
    setPage(0);
  };

  return (
    <div className="moduleContent">
      <Table>
        <TableHead>
          <TableRow>
            <TableCell>Date</TableCell>
            <TableCell>Amount</TableCell>
            <TableCell>Transaction Type</TableCell>
            <TableCell>Payee Type</TableCell>
            <TableCell>Status</TableCell>
            <TableCell>Memo</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {props.data
            .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
            .map(data => (
              <TableRow key={data.id}>
                <TableCell>{formatDate(data.transactionDate)}</TableCell>
                <TableCell>{formatCurrency(data.amount)}</TableCell>
                <TableCell>{formatDescription(data.typeDescription)}</TableCell>
                <TableCell>{data.payeeName}</TableCell>
                <TableCell>{formatStatusCode(data.status)}</TableCell>
                <TableCell>{data.memo}</TableCell>
              </TableRow>
            ))}
        </TableBody>
        <TableFooter>
          <TableRow>
            <TablePagination
              rowsPerPageOptions={[10, 20, 30, 50, 100]}
              colSpan={3}
              count={props.data.length}
              rowsPerPage={rowsPerPage}
              page={page}
              SelectProps={{
                inputProps: { "aria-label": "rows per page" },
                native: true
              }}
              onChangePage={handleChangePage}
              onChangeRowsPerPage={handleChangeRowsPerPage}
              ActionsComponent={TablePaginationActions}
            />
          </TableRow>
        </TableFooter>
      </Table>
    </div>
  );
}

const useStyles = makeStyles(theme => ({
  root: {
    flexShrink: 0,
    color: theme.palette.text.secondary,
    marginLeft: theme.spacing(2.5)
  }
}));
