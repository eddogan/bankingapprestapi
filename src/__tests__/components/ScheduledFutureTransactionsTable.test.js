import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import ScheduledFutureTransactionsTable from "../../components/ScheduledFutureTransactionsTable";

const filteredData = [];

it("renders correctly", () => {
  const tree = renderer
    .create(<ScheduledFutureTransactionsTable data={filteredData} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <ScheduledFutureTransactionsTable data={filteredData} />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});
