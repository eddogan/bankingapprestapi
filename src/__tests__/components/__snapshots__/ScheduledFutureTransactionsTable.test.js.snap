// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="moduleContent"
>
  <table
    className="MuiTable-root"
  >
    <thead
      className="MuiTableHead-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Date
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Amount
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Transaction Type
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Payee Type
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Status
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head"
          scope="col"
        >
          Memo
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root"
    />
    <tfoot
      className="MuiTableFooter-root"
    >
      <tr
        className="MuiTableRow-root MuiTableRow-footer"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTablePagination-root MuiTableCell-footer"
          colSpan={3}
        >
          <div
            className="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
          >
            <div
              className="MuiTablePagination-spacer"
            />
            <span
              className="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
            >
              Rows per page:
            </span>
            <div
              className="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
              onClick={[Function]}
            >
              <select
                aria-label="rows per page"
                className="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiInputBase-input MuiInputBase-inputSelect"
                multiple={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                value={10}
              >
                <option
                  className="MuiTablePagination-menuItem"
                  value={10}
                >
                  10
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={20}
                >
                  20
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={30}
                >
                  30
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={50}
                >
                  50
                </option>
                <option
                  className="MuiTablePagination-menuItem"
                  value={100}
                >
                  100
                </option>
              </select>
              <svg
                aria-hidden="true"
                className="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
                focusable="false"
                role="presentation"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 10l5 5 5-5z"
                />
              </svg>
            </div>
            <span
              className="MuiTypography-root MuiTablePagination-caption MuiTypography-caption MuiTypography-colorInherit"
            >
              0-0 of 0
            </span>
            <div
              className="makeStyles-root-99"
            >
              <button
                aria-label="first page"
                className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M18.41 16.59L13.82 12l4.59-4.59L17 6l-6 6 6 6zM6 6h2v12H6z"
                    />
                    <path
                      d="M24 24H0V0h24v24z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="previous page"
                className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="next page"
                className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
              <button
                aria-label="last page"
                className="MuiButtonBase-root MuiIconButton-root Mui-disabled Mui-disabled"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M5.59 7.41L10.18 12l-4.59 4.59L7 18l6-6-6-6zM16 6h2v12h-2z"
                    />
                    <path
                      d="M0 0h24v24H0V0z"
                      fill="none"
                    />
                  </svg>
                </span>
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
